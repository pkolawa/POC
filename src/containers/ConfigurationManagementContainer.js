import React from "react";
import { connect } from "react-redux";
import { withStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import Grid from "@material-ui/core/Grid";
import TreeNavigation from "../components/TreeNaviation/TreeNavigation";
import metaData from "./../components/TreeNaviation/meta-data-config.json";
import FormComponent from "./../components/FormComponent";
import Menu from "../components/Menu";
import Title from "../components/Title";

const styles = (theme) => ({
  root: {
    display: "flex",
    flexWrap: "wrap",
    flexDirection: "row",
  },
  toolbar: theme.mixins.toolbar,
  container: {
    flexBasis: `calc(100% - 240px)`,
  },
});

class ConfigurationManagementContainer extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      formData: [],
      // metaData: []
    };
  }

  componentDidMount() {
    //fetch('http://localhost:3001/configuration').then( metaData => this.setState({metaData})).catch(e => console.error(e))
  }

  handleNodeClick = (formData) => {
    if (formData && formData.length > 0) {
      this.setState({ formData });
    }
  };

  render() {
    const { classes } = this.props;
    const { formData } = this.state;

    return (
      <div className={classes.root}>
        <Menu />
        <Container className={classes.container}>
          <Title color="secondary" title="Configuration Page"/>
          <Grid container className={classes.gridContainer} spacing={3}>
            <Grid item xs={3}>
              <TreeNavigation
                treeItems={metaData}
                nodeClickHandler={this.handleNodeClick}
              />
            </Grid>
            <Grid item xs={9}>
              <FormComponent pageData={formData} />
            </Grid>
          </Grid>
        </Container>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  ...state,
});

const mapDispatchToProps = (dispatch) => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(styles)(ConfigurationManagementContainer));
