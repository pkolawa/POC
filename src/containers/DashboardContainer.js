import React from "react";
import { connect } from "react-redux";
import { withStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import DashboardCardList from "../components/DashboardCardList";

const styles = (theme) => ({});

class DashboardContainer extends React.Component {
  render() {
    return (
      <Container>
        <DashboardCardList />
      </Container>
    );
  }
}

const mapStateToProps = (state) => ({
  ...state,
});

const mapDispatchToProps = (dispatch) => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(styles)(DashboardContainer));
