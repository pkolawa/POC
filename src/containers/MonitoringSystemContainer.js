import { connect } from "react-redux";
import MonitoringSystem from "../components/Monitor"

const mapStateToProps = state => ({
    ...state
});

const mapDispatchToProps = dispatch => ({

});

const MonitoringSystemContainer = connect(
    mapStateToProps,
    mapDispatchToProps
)(MonitoringSystem);

export default MonitoringSystemContainer