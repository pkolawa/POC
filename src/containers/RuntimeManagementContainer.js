import React from "react";
import { connect } from "react-redux";
import { withStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import Menu from "../components/Menu";
import Title from "../components/Title";

const styles = (theme) => ({
  root: {
    display: "flex",
    flexWrap: "wrap",
    flexDirection: "row",
  },
  container: {
    flexBasis: `calc(100% - 240px)`,
  },
});

class RuntimeManagementContainer extends React.Component {
  render() {
    const { classes } = this.props;
    return (
      <div className={classes.root}>
        <Menu />
        <Container className={classes.container}>
          <Title color="secondary" title="Runtime Page" />
        </Container>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  ...state,
});

const mapDispatchToProps = (dispatch) => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withStyles(styles)(RuntimeManagementContainer));
