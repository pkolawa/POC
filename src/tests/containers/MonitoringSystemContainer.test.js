import React from "react";
import { shallow } from "enzyme";
import {MonitoringSystemContainer} from "../../containers/MonitoringSystemContainer";

test("Should Render Monitor Mgmt Container", () => {
  const wrapper = shallow(<MonitoringSystemContainer classes={{}}/>);
  expect(wrapper).toMatchSnapshot();
});