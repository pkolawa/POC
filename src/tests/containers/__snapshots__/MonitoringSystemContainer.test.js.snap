// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Monitor Mgmt Container 1`] = `
<div>
  <Menu />
  <WithStyles(ForwardRef(Container))>
    <div />
    <Title
      color="secondary"
      title="Monitor Page"
    />
    <WithStyles(SocketTest) />
  </WithStyles(ForwardRef(Container))>
</div>
`;
